### PR TITLE
feat: add /dogfood skill for automated release validation

### PR DESCRIPTION
**Pull request by carlos-alm**
## Summary
- Adds a `/dogfood <version>` skill that runs a full 10-phase validation session against a published npm release
- Installs from npm, verifies native binary, tests all 22+ CLI commands with flags/edge cases, tests rebuild/staleness/embedding consistency, compares WASM vs native engines, runs changelog-targeted tests, auto-files bugs as GitHub issues with fix PRs, and generates a structured report
- Self-improving: the skill encourages the agent to update its own instructions based on findings

## Test plan
- [ ] Run `/dogfood 2.2.1` and verify the skill executes end-to-end
- [ ] Verify report is written to `generated/DOGFOOD_REPORT_v2.2.1.md`

### CODE REVIEW

File: .claude/skills/dogfood/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+
+1. In the codegraph repo (not the temp dir), edit `package.json` `optionalDependencies` to pin all `@optave/codegraph-*` packages to `$ARGUMENTS`.
+2. Run `npm install` to update the lockfile.
+3. Commit and push directly to main:
+   ```bash
+   git checkout main
+   git add package.json package-lock.json
+   git commit -m "chore: pin native binaries to v$ARGUMENTS after clean dogfood"
+   git push origin main
+   ```
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply .claude/skills/dogfood/SKILL.md]: Fixed in beddf94 — now creates a PR (`chore/pin-native-binaries-v<version>`) instead of pushing directly to main.
